app/controllers/FeedbackFormsReportController.scala

```diff
@@ -43,52 +45,45 @@ class FeedbackFormsReportController @Inject()(messagesApi: MessagesApi,
   implicit val userFeedbackReportFormat: OFormat[UserFeedbackResponse] = Json.format[UserFeedbackResponse]
   implicit val feedbackReportHeaderFormat: OFormat[FeedbackReportHeader] = Json.format[FeedbackReportHeader]
   implicit val feedbackReportFormat: OFormat[FeedbackReport] = Json.format[FeedbackReport]
+  implicit val reportResult: OFormat[ReportResult] = Json.format[ReportResult]
 
   def renderUserFeedbackReports: Action[AnyContent] = userAction.async { implicit request =>
-    generateSessionFeedbackReport(Some(request.user.email)).map { reportInfo =>
-      Ok(views.html.reports.myreports(reportInfo))
-    }
+      Future.successful(Ok(views.html.reports.myreports()))
   }
 
   def renderAllFeedbackReports: Action[AnyContent] = adminAction.async { implicit request =>
-    generateSessionFeedbackReport(None).map { reportInfo =>
-      Ok(views.html.reports.allreports(reportInfo))
-    }
+      Future.successful(Ok(views.html.reports.allreports()))
   }
 
-  private def generateSessionFeedbackReport(presenter: Option[String]): Future[List[FeedbackReportHeader]] = {
-    presenter.fold {
-      val activeSessions = sessionsRepository.activeSessions(None)
-      val sessionTillNow = sessionsRepository.userSessionsTillNow(None)
-      generateReport(activeSessions, sessionTillNow)
-    } { email =>
-      val presenterActiveSessions = sessionsRepository.activeSessions(Some(email))
-      val presenterSessionTillNow = sessionsRepository.userSessionsTillNow(Some(email))
-      generateReport(presenterActiveSessions, presenterSessionTillNow)
+  def manageAllFeedbackReports(pageNumber: Int): Action[AnyContent] = adminAction.async { implicit request =>
+    val sessionTillNow = sessionsRepository.userSessionsTillNow(None, pageNumber)
+    generateReport(sessionTillNow).flatMap { reportInfo =>
+      sessionsRepository
+        .activeCount(None)
```

Comment:
Cross check if cancelled can be added to selector of activeCount else create a separate method.